### ISSUE TITLE
bug(MatIcon): Aria label on material icons is ignored by ChromeVox screen reader

### ISSUE THREAD

**Issue by alicequiros** [angular/components#25147]
### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### The previous version in which this bug was not present was

N/A

### Description

ChromeVox does not include the aria-label on material icons (both basic and svg) in the document flow. This makes it difficult for users who prefer to use a screen reader to identify icons on a page.

### Reproduction

Steps to reproduce:
0. On a Chromebook, open Chrome and enable ChromeVox
1. Navigate to the demos at https://material.angular.io/components/icon/examples
2. Use tab and search+arrow keys to navigate to the Basic Icon and the SVG Icon on the page
3. Observe that CVox reads "home" for the basic icon rather than its aria-label, and ignores the SVG icon completely

### Expected Behavior

The aria-label on each icon should be included in the document flow. Instead of "home", I expect ChromeVox to read out the full aria label of the icon (which is "Example home icon"). It should also do the same for the SVG icon on the same page (which is "Example thumbs up SVG icon").

### Actual Behavior

Icon labels are left out of the document flow, and it's left to the user to use Search+G to get any information about them. However, the "home" text from the basic icon is read out in the basic document flow (this is not its aria label). 

### Environment

- Browser(s): Chrome v102.0.5005.75 (Official Build) (64-bit)
- Operating System (e.g. Windows, macOS, Ubuntu): ChromeOS v102.0.5005.75 (Official Build) (64-bit)

**Comment by zarend** [comment]
Ty for reporting. We might need to wrap the text content inside the icon with aria-hidden= true, then test that it doesn't break NVDA/Jaws.

For accessibility, a great way to convey the label is with a sibling span next to the icon.
```
<span class="cdk-visually-hidden">Home screen</span>
<mat-icon>home</mat-icon>
```


For further documentation on a11y best practices for icon: https://material.angular.io/components/icon/overview#accessibility

**Comment by alicequiros** [comment]
I agree with the aria-hidden wrapper. I also agree that adding a hidden span works well, but it's also not entirely feasible to add a span alongside every icon on an existing website.

Since this also seems to be a Chromevox issue (VoiceOver, NVDA, and JAWS have no problem reading the icons), I filed an identical bug with the ChromeOS team [via crbug](https://bugs.chromium.org/p/chromium/issues/detail?id=1338979)

**Comment by calebegg** [comment]
> There really is not a situation in which a `<mat-icon>` should have an `aria-label`.

I guess if most screen readers support it, and the last one has plans to....why not? It seems like a saner way to add alt text to an icon compared to `cdk-visually-hidden`, which always seems like a hack to me....

**Comment by jelbourn** [comment]
@calebegg it's perhaps nitpicky, but icons shouldn't have alt text because they are almost never content images. `role="image"` with alt text is a different experience from a text node. In places where people are using an indicator icon pattern, I believe that a text node is the best user experience in a screen reader because it doesn't include any superfluous noise. 

That said, maybe `MatIcon` could potentially expose an `@Input` for expressing this text more concisely. I'm iffy about that, though, since it's only really relevant to one of the icon usage patterns.

**Comment by amysorto** [comment]
Closing this issue due to no recent activity, if this is still an issue please create a new issue with a reproduction.
